gobblin-utility/src/main/java/org/apache/gobblin/util/HadoopUtils.java

```diff
@@ -757,6 +761,57 @@ public static void safeRenameRecursively(FileSystem fileSystem, Path from, Path
     }
   }
 
+  public static void ensureDirectoryExists(FileSystem fs, Path path, Iterator<OwnerAndPermission> ownerAndPermissionIterator)
+      throws IOException {
+
+    if (fs.exists(path)) {
+      return;
+    }
+
+    if (ownerAndPermissionIterator.hasNext()) {
+      OwnerAndPermission ownerAndPermission = ownerAndPermissionIterator.next();
+
+      if (path.getParent() != null) {
+        ensureDirectoryExists(fs, path.getParent(), ownerAndPermissionIterator);
+      }
+
+      if (!fs.mkdirs(path)) {
+        // fs.mkdirs returns false if path already existed. Do not overwrite permissions
+        return;
+      }
+
+      List<AclEntry> aclEntries = ownerAndPermission.getAclEntries();
+      if (!aclEntries.isEmpty()) {
+        // use modify acls instead of setAcl since latter requires all three acl entry types: user, group and others
+        // while overwriting the acls for a given path. If anyone is absent it fails acl transformation validation.
+        fs.modifyAclEntries(path, aclEntries);
+      }
+
+      if (ownerAndPermission.getFsPermission() != null) {
+        log.debug("Applying permissions {} to path {}.", ownerAndPermission.getFsPermission(), path);
+        fs.setPermission(path, addExecutePermissionToOwner(ownerAndPermission.getFsPermission()));
+      }
+
+      String group = ownerAndPermission.getGroup();
+      String owner = ownerAndPermission.getOwner();
+      try {
+        if (group != null || owner != null) {
+          log.debug("Applying owner {} and group {} to path {}.", owner, group, path);
+          fs.setOwner(path, owner, group);
+        }
+      } catch (IOException ioe) {
+        log.warn("Failed to set owner and/or group for path " + path + " to " + owner + ":" + group, ioe);
+      }
```

Comment:
nit: streamline the `try` ... `catch` -
```
String owner = ...
if (group != null || owner != null) {
  log.debug(...);
  try {
    fs.setOwner(...);
  } catch (IOException ioe) {
  ...
}
```
also, I know logging but otherwise swallowing the exception is existing, b/w-compat behavior - but is it what we want?

Reply:
I will add a configuration to handle the failure case but default to false (allow failures and only log) if it does fail.